src/resources/HttpResource.js

```diff
@@ -0,0 +1,69 @@
+/* @flow */
+
+import type HttpApi, { Args, Data } from '../api/HttpApi';
+import type { Context } from '../types/NodeType';
+import defaultUrlJoin from '../utils/urlJoin';
+import Resource from './Resource';
+
+export type Endpoint = string | ((id?: string) => string);
+
+export type HttpResourceOptions = {
+  endpoint: Endpoint,
+  urlJoin?: (...paths: Array<?string>) => string,
+};
+
+export default class HttpResource extends Resource {
+  api: HttpApi;
+  options: HttpResourceOptions;
+  _endpoint: Endpoint;
+  _urlJoin: (...paths: Array<?string>) => string;
```

Comment:
what do you think of leaving out the `urlJoin` here for now and just using the "dumb" joining we already have? i know it's not perfect but it feels weird right now to have `PaginatedHttpResource` be a subclass, but for URL joining to be a ctor arg